Client/src/components/contanier/staff/Student/AddStudent/index.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react'
+import LabeledInput from './../../../../common/LabeledInput'
+import Button from './../../../../common/Button';
+import { addStudentSchema } from './../../../../../helpers/validation-schema'
+import './index.css';
+let valid;
+
+export default class AddStudent extends Component {
+  state = {
+    firstname: '',
+    address: '',
+    level: 0,
+    phonenumber: '',
+    lastname: '',
+    mobilenumber: '',
+    password: '',
+    Error: {}
+  }
+  handleInput = ({ target: { value, name } }) => this.setState({ [name]: value.trim() });
+
+  handleClick = (e) => {
+    e.preventDefault();
+    this.setState({ Error: '' })
+    addStudentSchema
+      .validate({ ...this.state }, {
+        abortEarly: false
+      })
+      .then((value) => {
+        // right fetch here 
+      })
+      .catch(({ inner }) => {
```

Comment:
@Fatmasiam It might be a good idea to rearrange these promise handlers slightly, to move the catch() in front of the then():
```
addStudentSchema
  .validate()
  .catch()
  .then()
```
This is because you will soon add your fetch call inside then(), and you don't want this catch block to also be handling the fetch errors (because those won't have `inner`, they need slightly different error handling).  By moving your catch() above the then(), that catch will only be responsible for validation errors, and later you could add another catch() underneath the fetch to handle those errors.  Let me know if that makes sense or is unclear!